File: backend/Core/Models/Dtos/LookupRequestDto.cs
using Core.Enums;

namespace Core.Models.Dtos
{
    public class LookupRequestDto
    {
        public string? SearchText { get; set; }
        //public int? MaxResults { get; set; } = 1000;
        //public bool? OnlyVisible { get; set; }           // e.g. only windows with MainWindowTitle
        public List<int> ExcludedIds { get; set; } = new List<int>();

        // Scope filters. Lookup.flowLocation uses FlowId, Lookup.flowStep walks up from FlowStepId.
        public int? FlowId { get; set; }
        public int? FlowStepId { get; set; }

        // Lookup.flowSearchArea only: window steps want APPLICATION areas, not monitors.
        public FlowSearchAreaTypeEnum? FlowSearchAreaType { get; set; }
    }
}

File: frontend/src/shared/models/lazy-data/lookup-request.dto.ts
import type { FlowAreaTypeEnum } from "@/shared/enums/backend/flow-area-type.enum";
import type { StepResultKindEnum } from "@/shared/enums/backend/step-result-kind-enum";

export interface LookupRequestDto {
  searchText?: string;
  excludedIds?: number[];

  // Scope filters. Lookup.flowPoint uses flowId, Lookup.flowStep walks up from flowStepId.
  flowId?: number;
  flowStepId?: number;

  // Lookup.flowArea only.
  flowAreaType?: FlowAreaTypeEnum;

  // Lookup.flowStep only: a cursor step wants a location, a condition wants a value.
  resultKind?: StepResultKindEnum;
}
